src/debug.sh

```diff
@@ -0,0 +1,504 @@
+#   enables a list pattern.
+# @events_str User input that needs to be parsed.
+#
+# Return:
+# In case of an error, returns EINVAL (22).
+function convert_event_syntax_to_sys_path_hash()
+{
+  local option="$1"
+  shift
+  local events_str="$*"
+
+  IFS=';' read -r -a events_str <<< "$events_str"
+  # Target event. e.g.: amdgpu_dm or amdgpu
+  for event in "${events_str[@]}"; do
+    if [[ -n "$option" ]]; then
+      events_hash["$EVENT_BASE_PATH/$event"]=''
+      continue
+    fi
+
+    if [[ "$event" =~ .*':'.* ]]; then
+      root_event=$(cut -d ':' -f1 <<< "$event")
+      specific_events_str=$(cut -d ':' -f2 <<< "$event")
+      # Invalid syntax
+      if [[ -z "$specific_events_str" ]]; then
+        return 22 # EINVAL
+      fi
+    elif [[ -n "$event" && ! "$event" =~ .*','.* ]]; then
+      events_hash["$EVENT_BASE_PATH/$event"]=''
+      continue
+    else
+      return 22 # EINVAL
+    fi
+
+    # Check specific event
+    IFS=',' read -r -a specific_events <<< "$specific_events_str"
+    for specific_event in "${specific_events[@]}"; do
+      local specific_filter=''
+      local hash_key=''
+
+      # Check for event filter
+      if [[ "$specific_event" =~ .*'['.*']'.* ]]; then
+        specific_filter=${specific_event%]*}
+        specific_filter=${specific_filter##*[}
+        specific_event=$(cut -d "[" -f1 <<< "$specific_event")
+      fi
+
+      hash_key="$EVENT_BASE_PATH/$root_event/$specific_event"
+      events_hash["$hash_key"]="$specific_filter"
+    done
+  done
+
+  #if [[ "$option" == "$LIST_OPTION" && -z "$event" ]]; then
+  if [[ -z "$event" ]]; then
+    events_hash["$EVENT_BASE_PATH"]=''
+  fi
+}
+
+function parser_debug_options()
+{
+  local options
+  local short_options
+  local long_options
+  local transition_variables
+
+  long_options='remote:,event:,cmd:,local,history,disable,list,follow,help'
+  short_options='f,h,e,c,h,d,l,k'
+
+  options="$(getopt -q --options "$short_options" \
+    --longoptions "$long_options" \
+    -- "$@")"
+
+  if [[ "$?" != 0 ]]; then
+    options_values['ERROR']="$(getopt --name 'debug' --options "$short_options" \
+      --longoptions "$long_options" \
+      -- "$@" 2>&1 > /dev/null)" # Get errors from stderr
+    return 22                    # EINVAL
+  fi
+
+  options_values['TARGET']=''
+  options_values['EVENT']=''
+  options_values['CMD']=''
+  options_values['HISTORY']=0
+  options_values['DISABLE']=0
+  options_values['LIST']=''
+  options_values['FOLLOW']=0
+
+  # Set default values
+  if [[ -n ${configurations[default_deploy_target]} ]]; then
+    transition_variables=${configurations[default_deploy_target]}
+    options_values['TARGET']=${deploy_target_opt[$transition_variables]}
+    # kw does not support this feature on turn-off vm
+    if [[ "${options_values['TARGET']}" == 1 ]]; then
+      complain 'debug is not supported in VM mode, please change default_deploy_target'
+      exit 95 # EOPNOTSUPP
+    fi
```

Comment:
This implies that if I set `default_deploy_target` to be `vm`, and then run `kw debug --remote` ... or even `kw debug -h`, I'll get a complaint saying that kw debug doesn't support VM.